File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int leftSum = 0;
        int total = 0;
        for(int i=0; i<nums.size(); i++){
            total += nums[i];
        }
        
        for(int i=0; i<nums.size(); i++){
            int rightSum = total - leftSum - nums[i];
            if(rightSum == leftSum) return i;
            
            leftSum = leftSum + nums[i];
        }
        
        return -1;
    }
};

File: 0724-find-pivot-index/0724-find-pivot-index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total = sum(nums)
        left = 0
        
        for i in range(0, len(nums)):
            right = total - left - nums[i]
            if(right == left):
                return i
            left += nums[i]
        
        return -1
                